### DISCUSSION TITLE
Templ: command not found

### DISCUSSION THREAD

**Discussion by lFourl** [a-h/templ#705] [General]
Hi All,

I have recently deployed a Templ app onto an EC2 instance. It is currently working, but I need to add templ generate to my Dockerfile. However when I try to do a `templ generate` from the terminal, I am getting a command not found. I have tried to install templ into my app directory with no luck using `go install github.com/a-h/templ/cmd/templ@latest` and I am currently able to run `templ generate` locally, but I am not sure what I am missing. Any thoughts as to what I am missing or how Templ should properly be installed on an amazonlinux2 server?

**Comment by a-h** [comment]
I'm not really sure what your workflow is here. You're talking about both Docker and installing commands into an EC2 instance.

Generally, you'd use `templ generate` and commit the generated files into source control. Then, you build your app using CI/CD, and copy the resulting app into the Docker container. The templ binary isn't required at runtime. 

Perhaps you could explain further, and provide examples?

**Reply by lFourl** [reply]
This is extremely helpful and makes sense. I was imagining adding `templ generate` to my dockerfile and having it run during each build, but the workflow that you mentioned makes a lot more sense. Thanks for the help!
